### ISSUE TITLE
brew upgrade swiftlint - fails

### ISSUE THREAD

**Issue by burczyk** [realm/SwiftLint#2240]
Upgrading swiftlint with Homebrew (`brew upgrade swiftlint`) fails. Please see a log below. Is it my environment issue or can you help me?

```
==> Upgrading 1 outdated package, with result:
swiftlint HEAD-755f3ad -> HEAD-755f3ad
==> Upgrading swiftlint
==> Cloning https://github.com/realm/SwiftLint.git
Updating /Users/kamil/Library/Caches/Homebrew/swiftlint--git
==> Checking out branch master
Synchronizing submodule url for 'Carthage/Checkouts/Nimble'
Synchronizing submodule url for 'Carthage/Checkouts/Quick'
Synchronizing submodule url for 'Carthage/Checkouts/Result'
Synchronizing submodule url for 'Carthage/Checkouts/xcconfigs'
Synchronizing submodule url for 'Externals/Nimble'
Synchronizing submodule url for 'Carthage/Checkouts/Commandant'
Synchronizing submodule url for 'Carthage/Checkouts/Result'
Synchronizing submodule url for 'Carthage/Checkouts/SWXMLHash'
Synchronizing submodule url for 'Carthage/Checkouts/Yams'
Synchronizing submodule url for 'Carthage/Checkouts/xcconfigs'
Synchronizing submodule url for 'Carthage/Checkouts/Nimble'
Synchronizing submodule url for 'Carthage/Checkouts/Quick'
Synchronizing submodule url for 'Carthage/Checkouts/Result'
Synchronizing submodule url for 'Carthage/Checkouts/xcconfigs'
Synchronizing submodule url for 'Externals/Nimble'
error: Server does not allow request for unadvertised object 89728970c0a244c43f29c4155a1b1dc20b3c4633
Fetched in submodule path 'Carthage/Checkouts/SwiftyTextTable', but it did not contain 89728970c0a244c43f29c4155a1b1dc20b3c4633. Direct fetching of that commit failed.
Error: Failed to download resource "swiftlint"
Failure while executing: git submodule -q update --init --recursive
```

**Comment by jpsim** [comment]
Can you share steps to reproduce? I'm not seeing this when building from source:

```
$ which swiftlint
swiftlint not found
$ brew install --build-from-source swiftlint
==> Cloning https://github.com/realm/SwiftLint.git
Cloning into '/Users/jsimard/Library/Caches/Homebrew/swiftlint--git'...
remote: Counting objects: 422, done.
remote: Compressing objects: 100% (378/378), done.
remote: Total 422 (delta 71), reused 99 (delta 29), pack-reused 0
Receiving objects: 100% (422/422), 739.97 KiB | 1.25 MiB/s, done.
Resolving deltas: 100% (71/71), done.
Note: checking out '8c5d2afa7c8f58bb8302cc5cc679b64741490982'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

==> Checking out tag 0.26.0
==> make prefix_install PREFIX=/usr/local/Cellar/swiftlint/0.26.0 TEMPORARY_FOLDER=/private/tmp/swiftlint-20180624-7756-1v967sl/SwiftLint.dst
🍺  /usr/local/Cellar/swiftlint/0.26.0: 6 files, 15MB, built in 2 minutes 26 seconds
$ swiftlint version
0.26.0
```

**Comment by burczyk** [comment]
Basically, I have a swiftlint in version 0.24.0 and I'm trying to update it with brew.

```bash
$ swiftlint version
0.24.0

$ brew upgrade swiftlint -v
==> Upgrading 1 outdated package, with result:
swiftlint HEAD-755f3ad -> HEAD-755f3ad
==> Upgrading swiftlint
rm /usr/local/bin/swiftlint
/usr/bin/sandbox-exec -f /tmp/homebrew20180626-60305-1a1qj7l.sb nice /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/build.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/swiftlint.rb --verbose --HEAD
==> Cloning https://github.com/realm/SwiftLint.git
Updating /Users/kamil/Library/Caches/Homebrew/swiftlint--git
git config remote.origin.url https://github.com/realm/SwiftLint.git
git config remote.origin.fetch +refs/heads/master:refs/remotes/origin/master
git fetch origin
==> Checking out branch master
git checkout -f master --
Already on 'master'
Your branch is up-to-date with 'origin/master'.
git reset --hard origin/master
HEAD is now at 1c8d5a49 Merge pull request #2257 from marcelofabri/format-dont-touch
git submodule foreach --recursive git submodule sync
Entering 'Carthage/Checkouts/Commandant'
Synchronizing submodule url for 'Carthage/Checkouts/Nimble'
Synchronizing submodule url for 'Carthage/Checkouts/Quick'
Synchronizing submodule url for 'Carthage/Checkouts/Result'
Synchronizing submodule url for 'Carthage/Checkouts/xcconfigs'
Entering 'Carthage/Checkouts/Commandant/Carthage/Checkouts/Nimble'
Entering 'Carthage/Checkouts/Commandant/Carthage/Checkouts/Quick'
Synchronizing submodule url for 'Externals/Nimble'
Entering 'Carthage/Checkouts/Commandant/Carthage/Checkouts/Quick/Externals/Nimble'
Entering 'Carthage/Checkouts/Commandant/Carthage/Checkouts/Result'
Entering 'Carthage/Checkouts/Commandant/Carthage/Checkouts/xcconfigs'
Entering 'Carthage/Checkouts/Result'
Entering 'Carthage/Checkouts/SWXMLHash'
Entering 'Carthage/Checkouts/SourceKitten'
Synchronizing submodule url for 'Carthage/Checkouts/Commandant'
Synchronizing submodule url for 'Carthage/Checkouts/Result'
Synchronizing submodule url for 'Carthage/Checkouts/SWXMLHash'
Synchronizing submodule url for 'Carthage/Checkouts/Yams'
Synchronizing submodule url for 'Carthage/Checkouts/xcconfigs'
Entering 'Carthage/Checkouts/SourceKitten/Carthage/Checkouts/Commandant'
Synchronizing submodule url for 'Carthage/Checkouts/Nimble'
Synchronizing submodule url for 'Carthage/Checkouts/Quick'
Synchronizing submodule url for 'Carthage/Checkouts/Result'
Synchronizing submodule url for 'Carthage/Checkouts/xcconfigs'
Entering 'Carthage/Checkouts/SourceKitten/Carthage/Checkouts/Commandant/Carthage/Checkouts/Nimble'
Entering 'Carthage/Checkouts/SourceKitten/Carthage/Checkouts/Commandant/Carthage/Checkouts/Quick'
Synchronizing submodule url for 'Externals/Nimble'
Entering 'Carthage/Checkouts/SourceKitten/Carthage/Checkouts/Commandant/Carthage/Checkouts/Quick/Externals/Nimble'
Entering 'Carthage/Checkouts/SourceKitten/Carthage/Checkouts/Commandant/Carthage/Checkouts/Result'
Entering 'Carthage/Checkouts/SourceKitten/Carthage/Checkouts/Commandant/Carthage/Checkouts/xcconfigs'
Entering 'Carthage/Checkouts/SourceKitten/Carthage/Checkouts/Result'
Entering 'Carthage/Checkouts/SourceKitten/Carthage/Checkouts/SWXMLHash'
Entering 'Carthage/Checkouts/SourceKitten/Carthage/Checkouts/Yams'
Entering 'Carthage/Checkouts/SourceKitten/Carthage/Checkouts/xcconfigs'
Entering 'Carthage/Checkouts/SwiftyTextTable'
Entering 'Carthage/Checkouts/Yams'
Entering 'Carthage/Checkouts/xcconfigs'
git submodule update --init --recursive
error: Server does not allow request for unadvertised object 89728970c0a244c43f29c4155a1b1dc20b3c4633
Fetched in submodule path 'Carthage/Checkouts/SwiftyTextTable', but it did not contain 89728970c0a244c43f29c4155a1b1dc20b3c4633. Direct fetching of that commit failed.
Error: Failed to download resource "swiftlint"
Failure while executing: git submodule update --init --recursive
ln -s ../Cellar/swiftlint/HEAD-755f3ad/bin/swiftlint swiftlint
```

**Comment by burczyk** [comment]
Uninstalling:

`brew uninstall swiftlint --force` 

and then installing again:

`brew install swiftlint`

solved the issue. Apparently there was something wrong with the update process.

**Comment by marcelofabri** [comment]
It seems that you were able to resolve the issue. Closing this. Feel free to reopen if it's still a problem.
